types/index.d.ts

```diff
@@ -181,7 +181,7 @@ declare module "miragejs/-types" {
   }
 
   type WithFactoryMethods<T> = {
-    [K in keyof T]: T[K] | ((n: number) => T[K]);
+    [K in keyof T]: T[K] | ((this: FlattenFactoryMethods<T>, n: number) => T[K]);
```

Comment:
What exactly does this do? 

As I understand it, it takes a factory and translates method return types into basic types, so 

```js
{
    age: number;
    height: () => string;
}
```

becomes

```js
{
    age: number;
    height: string;
}
```